### PR TITLE
Initial setup.py version and updated Makefile

### PR DESCRIPTION
**Pull request by alexonea**
This change adds python specific package distribution and build using ```setup.py```.
The files involved in providing functionalities such as pip support are:

* ```setup.py```    (main configuration script)
* ```setup.cfg```   (specifications for wheel and license)
* ```MANIFEST.in```

In addition to the above, the following adaptations have been made to accustom
the new ```setup.py``` infrastructure:

* ```iasimage``` has been renamed to``` iasimage.py``` (needed by ```setup.py```)
* a link with the old name (```iasimage```) has been created to ```iasimage.py``` for backwards compatibility
* now the tests directory contains an ```__init__.py``` file for automatic tests discovery by ```setup.py```
* the ```.travis.yml``` CI configuration has been updated to also perform tests through ```setup.py```
* ```Makefile``` has been updated to reflect the rename of ```iasimage``` to ```iasimage.py``` but keeping the installed file with the old ```iasimage``` name.
* ```tests/test_iasimage.py``` has been updated to start the ```iasimage.py``` script using the python interpreter version mandated by ```/usr/bin/env``` instead of just ```python```. This way we make sure that tests are run within the same environment as the ```iasimage``` script itself after install.

Also, a package has been uploaded the test PyPI repository [here](https://test.pypi.org/project/iasimage/). You can also install the test package using the following command:
```
pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple iasimage
```
Maintainer information has been left out for now.

Let me know what you think!

### CODE REVIEW

**Comment by huang-jin-intel** [comment]
Alex, 

Awesome work and thank you. This script is also supposed to run on Windows. How does /usr/bin/env work for both python 2 and 3?

Aaron

File: tests/test_iasimage.py [hunk 1]
```diff
@@ -25,11 +26,11 @@ def tearDown(self):
 				pass
 
 	def test_help(self):
-		cmd = ['python', IASIMAGE, '-h']
+		cmd = [ENV, 'python', IASIMAGE, '-h']
```
Comment (huang-jin-intel) [inline tests/test_iasimage.py]: I tested 'setup.py test' on Windows but it didn't work. I think the better way to handle this is to import iasimage as a module to run unit test for it, compared to calling iasimage as a script in the current code.

**Comment by alexonea** [comment]
@huang-jin-intel ```/usr/bin/env python``` starts whatever version of python it finds in ```PATH```. By doing the change we make sure that both the tests and ```iasimage``` are started from the same environment (same version of python since they are started using ```/usr/bin/env python``` (otherwise python might be a link or an alias and we cannot guarantee that they are started the same).

```/usr/bin/env``` does not implicitly make the script work for both versions of python. It defaults to ```python2.7```. However, the user can set-up a virtual environment (similar to what yocto does when builds a recipe in isolation) and now python ```/usr/bin/env python``` will start the right (desired) python version.

In addition, when installing ```iasimage``` using ```setup.py``` (or ```pip```), the executable script installed in the ```PATH``` is actually a wrapper which also can take care of the python version for you.

File: tests/test_iasimage.py (hunk 1 above)
Comment (alexonea) [reply tests/test_iasimage.py]: Is Windows support really required? I didn't see tests for Windows in CI, so I assumed that Windows is not a concern.

Nevertheless, I agree, importing the module makes more sense.

Comment (huang-jin-intel) [reply tests/test_iasimage.py]: We are using this tool in our boot loader project https://github.com/slimbootloader/slimbootloader. Many developers and users in firmware space are using Windows environment. Setting up a Windows CI would be a good idea. If you have some resources I am happy to consider / follow up.

**Comment by alexonea** [comment]
For the moment is it ok to just detect the OS in the test setup phase and
fallback to just python if on Windows?

Meanwhile I can think of ways to properly test this cross platform.

Alexandru N. Onea
Topor Computing SRL

**Comment by huang-jin-intel** [comment]
I am okay with the way it is now. The script is tested on Linux by Travis and work on both Windows and Linux.  We can improve other things overtime. 

Thank you!

**Comment by alexonea** [comment]
Thanks! 

Do you also want to register the package on PyPI so people (and build systems) can install it using ```pip```?

**Comment by huang-jin-intel** [comment]
I am not ready to publish it on PyPI unless we have regular release schedules. May I know what is your use case of this tool?

**Comment by alexonea** [comment]
I am using the tool for signing and creating IAS images to be loaded by an
automotive bootloader. I am working as part of a team developing
infotainment solutions for premium car OEMs.

My use of the tool is fully automated in the build system.

Alexandru N. Onea

**Pull request merged**